modules/lmdbbackend/lmdbbackend.cc

```diff
@@ -833,7 +833,10 @@ bool LMDBBackend::getDomainInfo(const DNSName &domain, DomainInfo &di, bool getS
       serFromString(val.get<string_view>(), rr);
 
       if(rr.content.size() >= 5 * sizeof(uint32_t)) {
-        uint32_t serial = *reinterpret_cast<uint32_t*>(&rr.content[rr.content.size() - (5 * sizeof(uint32_t))]);
+        uint32_t serial;
+        // a SOA has five 32 bit fields, the first of which is the serial
+        // there are two variable length names before the serial, so we calculate from the back
+        memcpy(&serial, &rr.content[rr.content.size() - (4 * sizeof(uint32_t))], sizeof(serial));
```

Comment:
How comes the 5 became a 4?

Reply:
ugh! I did that to see if it would actually break, and then I somehow committed that. I'll put on the brown paper bag now.

Reply:
Been there! It looks like it made its way into master first, then got backported, by the way.

Reply:
yes, just noticed that too